src/dotnet/CarbonAware.DataSources.WattTime/src/WattTimeDataSource.cs

```diff
@@ -63,10 +63,15 @@ private async Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(Location
 
         using (var activity = ActivitySource.StartActivity())
         {
-            BalancingAuthority balancingAuthority;
+            BalancingAuthority? balancingAuthority;
             try
             {
-                balancingAuthority = await this.LocationConverter.ConvertLocationToBalancingAuthorityAsync(location);
+                this.LocationSource.ToGeopositionLocation(location);
+                balancingAuthority = await WattTimeClient.GetBalancingAuthorityAsync(location.Latitude.ToString(), location.Longitude.ToString());
```

Comment:
It's unintuitive that we're accessing the original location with new fields. We should make the `ToGeopositionLocation` call should return a new object, not modify the original.

Reply:
I think you can use `!` to tell the compiler the field shouldn't be null

Reply:
Actually, looking at the code we do return a new Location object, not modifying the original so you need to use a new local variable in this function